discovery.go

```diff
@@ -1,78 +1,42 @@
 package discovery
 
 import (
-	"errors"
 	"fmt"
 	"net/url"
 	"os"
 	"strings"
-
-	kv "gopkg.in/clever/kayvee-go.v2"
 )
 
-// m is a convenience type for using kv.
-type m map[string]interface{}
-
 const (
-	templateVar = "SERVICE_%s_%s_%%s"
+	templateVar = "SERVICE_%s_%s_URL"
 )
 
-func getVar(envVar string) (string, error) {
-	envVar = strings.ToUpper(envVar)
-	envVar = strings.Replace(envVar, "-", "_", -1)
-	val := os.Getenv(envVar)
+// URLString finds the specified URL for a service based off of the service's name and which
+// interface you are accessing. Values are found in environment variables fitting the scheme:
+// SERVICE_{SERVICE NAME}_{INTERFACE NAME}_URL.
+func URLString(service, name string) (string, error) {
+	// build string
+	envKey := fmt.Sprintf(templateVar, service, name)
+
+	// standardize to uppercase & underscores
+	envKey = strings.Replace(strings.ToUpper(envKey), "-", "_", -1)
+
+	val := os.Getenv(envKey)
 	if val == "" {
-		return "", errors.New(kv.FormatLog("discovery-go", kv.Error, "missing env var", m{
-			"var": envVar,
-		}))
+		return "", fmt.Errorf("Missing env var, %s", envKey)
```

Comment:
Oops, I failed to put types on the struct. We could make all attributes either `string` or `url.URL`. Or we could even check what type they have in place and do the parsing for them.

```go
var ServiceConfig struct {
    InfluxDbURL string `discovery:"influxdb-service,protobuf"`
    Dependency2 string `discovery:"dependency-2,http"`
}

// this service is not given access to Dep1 or Dep2
var BrokenConfig struct {
    Dep1 string `discovery:"dep-1,http"`
    Dep2 string `discovery:"dep-1,http"`
}

func init () {
    err := discovery.Configure(&ServiceConfig)
    if err != nil {
        log.Fatal(err)
    }
    log.Printf("%#v\n", ServiceConfig)
    // {InfluxDbURL: tcp://influxdb-service.company.com Dependency2: http://dependency-2.company.com}

    err = discovery.Configure(&BrokenConfig)
    if err != nil {
        log.Fatal(err)
        // {source: "configuration", error: "No URL found for: ['dep-1', 'dep-2']"}
    }
}
```